tests/page-objects/default-mobile/search/search.wdio.page.js

```diff
@@ -1,14 +1,27 @@
 const searchPageDefault = require('@page-objects/default/search/search.wdio.page');
 
 const openSearchBox = () => $('.mm-search-bar-container .search-bar-left-icon .fa-search');
+const barcodeSearchInput = () => $('.barcode-scanner-input');
 
 const performSearch = async (term) => {
   await (await openSearchBox()).waitForClickable();
   await (await openSearchBox()).click();
   await searchPageDefault.performSearch(term);
 };
 
+const performBarcodeSearch = async (barcodeImagePath) => {
+  /*In this case the upload file button is hidden,
+  then we need to manipulate the DOM of the respective webelement to make it interactable.*/
+  browser.execute(function () {
+    document.getElementsByClassName('barcode-scanner-input')[0].style.display = 'block';
+  });
+  await (await barcodeSearchInput()).setValue(barcodeImagePath);
+  await browser.pause(1000);
```

Comment:
What is the reason for the 1s delay?

Reply:
We [experienced](https://github.com/medic/cht-core/issues/6669#issuecomment-1812666551) some delay when uploading and processing the barcode image. But I only tried the e2e test with the wait; It may be unnecessary.

Reply:
I try without the wait, and the test fails because it doesn't have time to process the search and show results. [This](https://github.com/medic/cht-core/blob/81f20d3ac93d9ec6b05addddfef4a8641f00d58e/tests/e2e/default-mobile/contacts/barcode-search-contacts.wdio-spec.js#L34) validation fails.